### PR TITLE
Refactoring to allow when condition plugins to not need p-m-d

### PR DESCRIPTION
**Pull request by abayer**
* JENKINS issue(s):
    * n/a
* Description:
    * Moving around various methods (and tweaking one of them to not need pipeline-stage-step) so that plugins like declarative-pipeline-when-conditions can avoid depending on pipeline-model-definition.
* Documentation changes:
    * n/a
* Users/aliases to notify:
    * @reviewbybees

### CODE REVIEW

File: pipeline-model-api/pom.xml [hunk 1]
```diff
@@ -62,7 +62,18 @@
...
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>script-security</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-cps</artifactId>
```
Comment (rsandell) [inline pipeline-model-api/pom.xml]: But now you are forcing users of api to get this one. Isn't that more bloated than forcing a dependency on declarative-api?

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy [hunk 2]
```diff
@@ -173,27 +172,12 @@ class Utils {
         }
     }
 
+    /**
+     * Use {@link ASTParserUtils#isStageWithOptionalName} instead.
```
Comment (rsandell) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy]: `@deprecated Use {@link ASTParserUtils#isStageWithOptionalName} instead.` will make javadoc generation handle it a bit smarter.

**Comment by abayer** [comment]
So fwiw, the idea here is to make it so that `pipeline-model-definition` can itself depend on `declarative-pipeline-when-conditions` in the future. I know more experimenting/fiddling is needed to actually get this to a point I'm happy with.

**Comment by abayer** [comment]
Giving up on this for now.

**Pull request closed without merging**